### PR TITLE
[CCXDEV-10763] Unit test for function fetchAllRulesContent

### PR DESCRIPTION
**Pull request by Jakub007d**
# Description

Unit tests for function fetchAllRulesContent

Fixes # [CCXDEV-10763](https://issues.redhat.com/browse/CCXDEV-10763)

## Type of change

Please delete options that are not relevant.

- Unit tests (no changes in the code)

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: differ/content_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+	"github.com/RedHatInsights/ccx-notification-service/conf"
+	"github.com/RedHatInsights/ccx-notification-service/types"
+	"github.com/RedHatInsights/insights-content-service/content"
+	"github.com/stretchr/testify/assert"
+)
+
+func makeFetchRequest(testFileName string) (err error, rules types.RulesMap) {
+	config, err := conf.LoadConfiguration(configFileEnvVariableName, defaultConfigFileName)
+	if err == nil {
```
Comment (tisnik) [inline differ/content_test.go:19]: idiomatic way is to use `if err != nil`, the "happy path" should be vertical

File: differ/content_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	allContentFromMap := getAllContentFromMap(rules)
+	assert.Empty(t, allContentFromMap)
+}
+func Handler(w http.ResponseWriter, r *http.Request) {
+	con, _, _ := content.ParseRuleContentDir("../" + r.RequestURI)
+	var temp bytes.Buffer
+	enc := gob.NewEncoder(&temp)
+	err := enc.Encode(con)
+	var tosend string
+	if err == nil {
```
Comment (tisnik) [inline differ/content_test.go:60]: dtto for idiomatic way to handle errors + "happy path"

File: differ/content_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		rules, err = fetchAllRulesContent(&depencenciesConfiguration)
+	}
+	return err, rules
+}
+func TestFetchAllRulesContentOk(t *testing.T) {
+	err, rules := makeFetchRequest("ok")
+	assert.NoError(t, err)
+	allContentFromMap := getAllContentFromMap(rules)
+	assert.NotEmpty(t, allContentFromMap)
+	assert.Len(t, allContentFromMap, 2)
```
Comment (tisnik) [inline differ/content_test.go:33]: I guess I myself will update tests to check the returned content more thoroughly later ;)

File: differ/content_test.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+}
+func TestFetchAllRulesContentNoInternal(t *testing.T) {
+	err, rules := makeFetchRequest("no_internal")
+	assert.NoError(t, err)
+	allContentFromMap := getAllContentFromMap(rules)
+	assert.NotEmpty(t, allContentFromMap)
```
Comment (tisnik) [inline differ/content_test.go:39]: I guess I myself will update tests to check the returned content more thoroughly later ;)

**Review by juandspy — APPROVED** [review]

**Review by tisnik — APPROVED** [review]
ok

**Review by joselsegura — APPROVED** [review]

**Pull request merged**